src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2167,6 +2167,197 @@ protected override void BeginProcessing()
     } // class NewServiceCommand
     #endregion NewServiceCommand
 
+    #region RemoveServiceCommand
+    /// <summary>
+    /// This class implements the remove-service command
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "Service", SupportsShouldProcess = true, DefaultParameterSetName = "Name")]
+    public class RemoveServiceCommand : ServiceBaseCommand
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Name of the service to remove
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "Name")]
+        [Alias("ServiceName", "SN")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The following is the definition of the input parameter "InputObject".
+        /// Specifies ServiceController object representing the services to be removed.
+        /// Enter a variable that contains the objects or type a command or expression
+        /// that gets the objects.
+        /// </summary>
+        [Parameter(ValueFromPipeline = true, ParameterSetName = "InputObject")]
+        public ServiceController InputObject { get; set; }
+
+        /// <summary>
+        /// The following is the definition of the input parameter "ComputerName".
+        /// Set the properties of service running on the list of computer names
+        /// specified. The default is the local computer.
+        /// Type the NETBIOS name, an IP address, or a fully-qualified domain name of
+        /// one or more remote computers. To indicate the local computer, use the
+        /// computer name, "localhost" or a dot (.). When the computer is in a different
+        /// domain than the user, the fully-qualified domain name is required.
+        /// </summary>
+        [Parameter(ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("cn")]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public String[] ComputerName { get; set; } = new string[] { "." };
+
+        #endregion Parameters
+
+        #region Overrides
+        /// <summary>
+        /// Remove the service
+        /// </summary>
+        [ArchitectureSensitive]
+        protected override void ProcessRecord()
+        {
+            ServiceController service = null;
+            string serviceComputerName = null;
+            foreach (string computer in ComputerName)
+            {
+                bool objServiceShouldBeDisposed = false;
+                try
+                {
+                    if (_ParameterSetName.Equals("InputObject", StringComparison.OrdinalIgnoreCase) && InputObject != null)
```

Comment:
Why are we using `_ParameterSetName` in the file (twice)? In all other files we use `ParameterSetName`.
I believe we should fix this.

Reply:
Looks like line 1604 used this and @joandrsn was just following this pattern.  Agree that we should change it to just `ParameterSetName` (including line 1604) for consistency.

Reply:
It seems we could remove `_ParameterSetName.Equals("InputObject", StringComparison.OrdinalIgnoreCase)` at all.

Reply:
I agree.  InputObject check should be sufficient